docs/src/main/sphinx/security/inspect-pem.rst

```diff
@@ -0,0 +1,139 @@
+=========
+PEM files
+=========
+
+PEM (Privacy Enhanced Mail) is a standard for public key and certificate
+information, and an encoding standard used to transmit keys and certificates.
+
+Trino supports PEM-encoded certificates. If you want to use other supported
+formats, see:
+
+*  :doc:`JKS keystores </security/inspect-jks>`
+*  For binary :ref:`PKCS 12 <glossPKCS12>` stores, look up alternate commands in
+   ``openssl`` references
+
+A single PEM-encoded file can contain either certificate or key pair
+information, or both in the same file. Certified keys can contain a chain of
+certificates from successive certificate authorities. If the certificate chain
+goes all the way back to a recognized Certificate Authority (CA), that cert does
+not need a separate truststore.
+
+Follow the steps in this topic to inspect and validate PEM-encoded key and
+certificate files. See :ref:`troubleshooting_keystore` to validate JKS keystores.
+
+.. _inspect_pems:
+
+Inspect PEM key files
+---------------------
+
+You may receive a single file that includes a private key and its certificate,
+or separate files. The file name extensions shown on this page are examples
+only; there is no extension naming standard.
+
+First, use the ``cat`` command to view these plain text files. For example:
+
+.. code-block:: shell
+
+    cat clustercoord.key
+    cat clustercoord.cert
+
+or:
+
+.. code-block:: shell
+
+    cat clustercoord.pem
+
+A key file looks something like the following:
+
+.. code-block:: text
+
+  -----BEGIN PRIVATE KEY-----
+  MIIEowIBAAKCAQEAwJL8CLeDFAHhZe3QOOF1vWt4Vuk9vyO38Y1y9SgBfB02b2jW
+  ....
+  -----END PRIVATE KEY-----
+
+If your key file reports ``BEGIN ENCRYPTED PRIVATE KEY`` instead, this means the
+key is encrypted and you must use the password to open or inspect the key. You
+may have specified the password when requesting the key, or the password could
+be assigned by your site's network managers.
+
+  If your key file reports ``BEGIN EC PRIVATE KEY`` or
+  ``BEGIN DSA PRIVATE KEY``, this designates keys using Elliptical Curve or
+  DSA alternatives to RSA. In these cases, consult ``openssl`` references for
+  the appropriate versions of the verification commands below.
+
+Inspect PEM certificate files
+-----------------------------
+
+A separate certificate file looks like the following example with the ``cat``
+command:
+
+.. code-block:: text
+
+  -----BEGIN CERTIFICATE-----
+  MIIDujCCAqICAQEwDQYJKoZIhvcNAQEFBQAwgaIxCzAJBgNVBAYTAlVTMRYwFAYD
+  ....
+  -----END CERTIFICATE-----
+  -----BEGIN CERTIFICATE-----
+  MIIDwjCCAqoCCQCxyqwZ9GK50jANBgkqhkiG9w0BAQsFADCBojELMAkGA1UEBhMC
+  ....
+  -----END CERTIFICATE-----
+
+The file can show a single certificate section, or more than one to express a
+chain of authorities, each certifying the previous.
+
+If you received a single file, make sure it shows at least one ``KEY`` and one
```

Comment:
I think we should instruct the user do this before inspecting, because this is what we need for Trino. Then in the `Inspect PEM` section we cover both certificates and private keys like we do in the JKS section.

Reply:
OK, I have now rearranged inspect-pem.rst to instruct concatenating separate key and cert files *first*. This required a rewrite to inspect and validate *sections* of a file, not files, with some simplification throughout. Check over the whole file to make sure I didn't miss anything.